### PR TITLE
Standardize SDK Logging, Phase 3-3

### PR DESCRIPTION
**Pull request by tkatzir**
## Related Issues
relates: https://jira-hq.paloaltonetworks.local/browse/CIAC-6259

### CODE REVIEW

File: demisto_sdk/commands/common/logger.py [hunk 1]
```diff
@@ -101,10 +104,168 @@ def set_demisto_logger(demisto_logger: logging.Logger):
...
+        logging.log(level_num, message, *args, **kwargs)
+
+    logging.addLevelName(level_num, level_name)
+    setattr(logging, level_name, level_num)
+    setattr(logging.getLoggerClass(), method_name, logForLevel)
+    setattr(logging, method_name, logToRoot)
+
+
+SUCCESS_LEVEL: int = 25
+if not hasattr(logging.getLoggerClass(), "success"):
```
Comment (ilaner) [inline demisto_sdk/commands/common/logger.py:157]: Shouldn't we do it in `logging_setup`?

File: demisto_sdk/commands/common/logger.py [hunk 2]
```diff
@@ -101,10 +104,168 @@ def set_demisto_logger(demisto_logger: logging.Logger):
...
+            current_escape = current_message[0 : current_message.find("]") + 1]
+            ret_value += current_escape
+            current_message = current_message[
+                len(current_escape) : current_message.find("]", len(current_escape)) + 1
+            ]
+
+        return ret_value
+
+    @staticmethod
+    def _insert_into_escapes(record: logging.LogRecord, string: str) -> str:
```
Comment (ilaner) [inline demisto_sdk/commands/common/logger.py:217]: Can you explain why we need it and what that does?

File: demisto_sdk/commands/common/logger.py [hunk 3]
```diff
@@ -199,6 +324,22 @@ def replace_escapes(self, message):
 
     set_demisto_logger(demisto_logger)
 
+    if notify_log_file_path and not current_log_file_path_notified:
+        import sys
```
Comment (ilaner) [inline demisto_sdk/commands/common/logger.py:328]: Why shouldn't we log this  anyway?

**Review by ilaner — COMMENTED** [review]
See comments

File: demisto_sdk/commands/common/logger.py (hunk 3 above)
Comment (tkatzir) [reply demisto_sdk/commands/common/logger.py]: [Done](https://github.com/demisto/demisto-sdk/pull/2994/commits/8c7e4b104f8fe9ec88c44caa46967884adfb93c8).

File: demisto_sdk/commands/common/logger.py (hunk 2 above)
Comment (tkatzir) [reply demisto_sdk/commands/common/logger.py:211]: To insert strings inside escaped messages, like when adding the logging level.

File: demisto_sdk/commands/common/logger.py (hunk 1 above)
Comment (tkatzir) [reply demisto_sdk/commands/common/logger.py]: [Done](https://github.com/demisto/demisto-sdk/pull/2994/commits/8c7e4b104f8fe9ec88c44caa46967884adfb93c8).

File: demisto_sdk/commands/common/logger.py [hunk 4]
```diff
@@ -101,6 +106,155 @@ def set_demisto_logger(demisto_logger: logging.Logger):
...
+        return ret_value
+
+    @staticmethod
+    def _insert_into_escapes(record: logging.LogRecord, string: str) -> str:
+        if not ColorConsoleFormatter._record_starts_with_escapes(record):
+            return string + record.getMessage()
+
+        # Need to "insert" the string into the escapes
+        start_escapes = ColorConsoleFormatter._get_start_escapes(record)
+        # import pdb; pdb.set_trace()
```
Comment (ilaner) [inline demisto_sdk/commands/common/logger.py:217]: remove comment

File: demisto_sdk/commands/common/logger.py [hunk 5]
```diff
@@ -163,24 +297,10 @@ def replace_escapes(self, message):
...
-        def format(self, record):
-            message = logging.Formatter.format(self, record)
-            message = self.replace_escapes(message)
-            return message
-
-        def replace_escapes(self, message):
-            for key in escapes:
-                message = message.replace(key, "")
-            return message
+    if os.getenv("DEMISTO_SDK_LOG_NO_COLORS", "False") == "True":
```
Comment (ilaner) [inline demisto_sdk/commands/common/logger.py:300]: use the `str2bool` function

File: demisto_sdk/commands/common/logger.py (hunk 4 above)
Comment (tkatzir) [reply demisto_sdk/commands/common/logger.py]: [Done](https://github.com/demisto/demisto-sdk/pull/2994/commits/3a4a797d3fc8e571ad5fe5f5fc18db19f92f9bd0).

File: demisto_sdk/commands/common/logger.py (hunk 5 above)
Comment (tkatzir) [reply demisto_sdk/commands/common/logger.py]: [Done](https://github.com/demisto/demisto-sdk/pull/2994/commits/00c68f386937f56f60f87547d69e4f58ace5ec78).

**Pull request merged**